components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -397,3 +405,86 @@ function OrgDefaultWorkspaceImageModal(props: OrgDefaultWorkspaceImageModalProps
         </Modal>
     );
 }
+
+interface WorkspaceClassOptionsProps {
+    settings: OrganizationSettings | undefined;
+    disabled: boolean;
+}
+const WorkspaceClassOptions = (props: WorkspaceClassOptionsProps) => {
+    const [errorMsg, setErrorMsg] = useState("");
+    const [selectedValue, setSelectedValue] = useState(props.settings?.allowedWorkspaceClasses ?? []);
+    const updateTeamSettings = useUpdateOrgSettingsMutation();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+    const handleUpdateTeamSettings = useCallback(
+        async (classes: string[]) => {
+            try {
+                await updateTeamSettings.mutateAsync(
+                    {
+                        ...props.settings,
+                        allowedWorkspaceClasses: classes,
+                    },
+                    {
+                        onSuccess: () => {
+                            toast({ message: "Available workspace classes updated." });
+                        },
+                    },
+                );
+            } catch (error) {
```

Comment:
Rather than catching this here and tracking our own `errorMsg` state we should use the `error` property that the `updateTeamSettings` hook result has, i.e. `updateTeamSettings.isError` && `updateTeamSettings.error`. This handles all of that state for us automatically.